classes/class-wpcom-liveblog-schema.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Class WPCOM_Liveblog_Schema
+ *
+ * Generate LiveBlogPosting Schema
+ * http://schema.org/LiveBlogPosting
+ *
+ */
+class WPCOM_Liveblog_Schema {
+
+	protected $query;
+	protected $post_id;
+	protected $permalink;
+
+
+	public function __construct( $post_id ) {
+		$this->query   = new WPCOM_Liveblog_Entry_Query( $post_id, WPCOM_Liveblog::key );
+		$this->post_id = $post_id;
+
+		// store permalink because it's used in a loop @ convert_comment_to_schema_entry
+		$this->permalink = get_permalink( $this->post_id );
+	}
+
+	public static function add_schema_to_content( $content ) {
+
+		if ( ! WPCOM_Liveblog::is_viewing_liveblog_post() ) {
+			return $content;
+		}
+
+		try {
+			$post_id = WPCOM_Liveblog::get_post_id();
+		} catch ( Exception $e ) {
+			$post_id = get_the_ID();
+		}
+
+		$schema = new static( $post_id );
```

Comment:
The use of static as a reference to itself was interesting. I had never seen it in use and was at first a bit confused. I looked it up and it's great and will work but I'm not sure what we gain vs the more traditional new WPCOM_Liveblog_Schema() or new self(); for example. Perhaps it's just me who hasn't had any experience with it but I think a comment here could be useful.

Reply:
`new static()` does the same thing as self, but is going to allow for extending the class much better. It's been around in PHP for quite some time, I think it's pretty safe to use. I can add a comment to clarify.